salt/utils/network.py

```diff
@@ -1881,7 +1883,28 @@ def dns_check(addr, port=80, safe=False, ipv6=None):
                     resolved = salt.utils.zeromq.ip_bracket(addr)
                     break
 
-                candidate_addr = salt.utils.zeromq.ip_bracket(h[4][0])
+                #Check for tcp transport in config opts
```

Comment:
I don't think we should load configs here. Let's instead accept `opts` as an argument to the function.

Reply:
Made suggested changes. Was originally modeling after https://github.com/saltstack/salt/blob/develop/salt/utils/http.py#L184.

Noticed a few other issues with using ipv6 and tcp.
Adjusted code in some new locations to account for zmq's bracket'ing.